packages/core/test/odata/v4/entity.spec.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  TestEntity,
+  TestEntityLvl2SingleLink,
+  TestEntityMultiLink,
+  TestEntitySingleLink
+} from '../../test-util/test-services/v4/test-service';
+
+describe('entity v4', () => {
+  let entity;
+
+  const state = {
+    stringProperty: '1',
+    collectionProperty: ['a', 'b'],
+    toSingleLink: {
+      toSingleLink: {
+        stringProperty: 'test'
+      },
+      customField: 2
+    },
+    toMultiLink: [
+      {
+        booleanProperty: false,
+        customField: 3
+      }
+    ],
+    customField: 1
+  };
+
+  beforeEach(() => {
+    entity = TestEntity.builder()
+      .stringProperty('1')
+      .collectionProperty(['a', 'b'])
+      .toSingleLink(
+        TestEntitySingleLink.builder()
+          .toSingleLink(
+            TestEntityLvl2SingleLink.builder().stringProperty('test').build()
+          )
+          .withCustomFields({
+            customField: 2
+          })
+          .build()
+      )
+      .toMultiLink([
+        TestEntityMultiLink.builder()
+          .booleanProperty(false)
+          .withCustomFields({
+            customField: 3
+          })
+          .build()
+      ])
+      .withCustomFields({
+        customField: 1
+      })
+      .build();
+  });
+  it('getCurrentMapKeys', () => {
+    const currentState = entity.getCurrentMapKeys();
```

Comment:
`getCurrentMapKeys()` returns a map.
`getCurrentMap` might be less confusing?

Reply:
This name is bad and I think it should be `getCurrentState`, but I didn't introduce this name and as it is protected it is basically public api 😕

Reply:
see my comment about the name above :-)